### PR TITLE
[chore] add helm chart update task to PR template

### PR DESCRIPTION
**Pull request by puckpuck**
Adds a task to the PR template to update the Helm chart, as well as additional comments on when docs and helm chart updates may be required.

### CODE REVIEW

**Comment by cartersocha** [comment]
Should we link to the actual repos?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,7 +8,19 @@ For new features contributions please make sure you have completed the following
 essential items:
 
 * [ ] `CHANGELOG.md` updated to document new feature additions
-* [ ] Appropriate documentation updates in the [docs][]
+* [ ] Appropriate documentation updates in the [docs repo]()
```
Comment (julianocosta89) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: This change will break the link.

**Review by julianocosta89 — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (julianocosta89) [reply .github/PULL_REQUEST_TEMPLATE.md:11]: At the bottom of the file we have the reference for the link:

`[docs]: https://opentelemetry.io/docs/demo/`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -8,7 +8,19 @@ For new features contributions please make sure you have completed the following
...
+* [ ] Appropriate Helm chart updates in the [helm charts repo]()
```
Comment (julianocosta89) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: We also need to add the reference for this link.

**Review by julianocosta89 — APPROVED** [review]

**Pull request merged**